### PR TITLE
Add new metrics for k8s 1.23 & 1.24

### PR DESCRIPTION
**Pull request by marcsanmi**
Please follow [conventional commit standards](https://www.conventionalcommits.org/en/v1.0.0/)
in your commit messages and pull request title.

## Give us some context

* What problems does this PR solve?
Adds documentation for new metrics introduced when supporting k8s v1.23 & 1.24
* Add any context that will help us review your changes such as testing notes,
  links to related docs, screenshots, etc.
https://github.com/newrelic/nri-kubernetes/issues/455
* If your issue relates to an existing GitHub issue, please link to it.

### CODE REVIEW

**Comment by marcsanmi** [comment]
cc/ @mangulonr

File: src/content/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data.mdx [hunk 1]
```diff
@@ -2060,6 +2072,16 @@ Query the `K8sContainerSample` event for container data:
       </td>
     </tr>
 
+    <tr>
+      <td>
+        `restartCountDelta`
+      </td>
+
+      <td>
+        Delta change of number of times the container has been restarted (requires k8s 1.23+).
```
Comment (sigilioso) [inline src/content/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data.mdx:2081]:
```suggestion
        Delta change of number of times the container has been restarted.
```

I think this metric does not require k8s 1.23+. Its value is got from `restartCount`.

File: src/content/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data.mdx [hunk 2]
```diff
@@ -2141,6 +2163,16 @@ Query the `K8sContainerSample` event for container data:
...
       </td>
     </tr>
+
+    <tr>
+      <td>
+        `containerOOMEventsDelta`
+      </td>
+
+      <td>
+        Number of out of memory events observed for the container (requires k8s v1.24+).
```
Comment (sigilioso) [inline src/content/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data.mdx:2173]:
```suggestion
        Number of "out of memory" events observed for the container (requires k8s v1.24+).
```
Quoting to group them together, I think it is clearer but I'm not sure if it is correct

File: src/content/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data.mdx [hunk 3]
```diff
@@ -2423,7 +2455,27 @@ Query the `K8sApiServerSample` [event](/docs/telemetry-data-platform/understand-
       </td>
 
       <td>
-        Number of stored objects at the time of last check, split by kind.
+        Number of stored objects at the time of last check, split by kind (alpha; deprecated in Kubernetes 1.22).
```
Comment (sigilioso) [inline src/content/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data.mdx:2458]: Maybe it is redundant but maybe it is also a bit clearer. Shall we mention here that it is replaced by `apiserverStorageObjects` in k8s 1.23+?

**Review by sigilioso — COMMENTED** [review]
Thanks for addressing this, I left some minor comments

File: src/content/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data.mdx (hunk 2 above)
Comment (marcsanmi) [reply src/content/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data.mdx]: Done in 0b782eae3fd7da1856370e2efed260a4fe70f300.

File: src/content/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data.mdx (hunk 3 above)
Comment (marcsanmi) [reply src/content/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data.mdx]: Done in 0b782eae3fd7da1856370e2efed260a4fe70f300.

**Review by sigilioso — APPROVED** [review]
👍

**Review by x8a — APPROVED** [review]
Thanks!

File: src/content/docs/release-notes/infrastructure-release-notes/kubernetes-integration-release-notes/kubernetes-integration-3-4-0.mdx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+subject: Kubernetes integration
+releaseDate: '2022-07-05'
+version: 3.4.0
+---
+
+### Added
+
+ - Add k8s v1.23 & v1.24 new metrics
```
[the remark this answers is not in the document]
Comment (marcsanmi) [reply src/content/docs/release-notes/infrastructure-release-notes/kubernetes-integration-release-notes/kubernetes-integration-3-4-0.mdx:9]: Done in https://github.com/newrelic/docs-website/pull/8412.

**Pull request merged**